mainhandler/vulnscan.go

```diff
@@ -94,143 +96,124 @@ func convertImagesToWebsocketScanCommand(registry *registryScan, sessionObj *uti
 	return webSocketScanCMDList
 }
 
-func (actionHandler *ActionHandler) getRegistryAuth(registryName string) (*registryAuth, error) {
-	secret, err := actionHandler.getRegistryScanSecret()
-	if err != nil {
-		return nil, err
-	}
-	secretData := secret.GetData()
-	var registriesAuth []registryAuth
-	registriesAuthStr, ok := secretData[registriesAuthFieldInSecret].(string)
-	if !ok {
-		return nil, fmt.Errorf("error parsing Secret: %s field must be a string", registriesAuthFieldInSecret)
-	}
-	data, err := base64.StdEncoding.DecodeString(registriesAuthStr)
-	if err != nil {
-		return nil, fmt.Errorf("error parsing Secret: %s", err.Error())
+func (actionHandler *ActionHandler) scanRegistries(sessionObj *utils.SessionObj) error {
+
+	registryScan := NewRegistryScan(actionHandler.k8sAPI)
+	var err error
+	if registryScan.isParseRegistryFromCommand(sessionObj.Command) {
+		err = registryScan.parseRegistryFromCommand(sessionObj)
+	} else {
+		err = registryScan.parseRegistryFromCluster(sessionObj)
 	}
-	registriesAuthStr = strings.Replace(string(data), "\n", "", -1)
-	err = json.Unmarshal([]byte(registriesAuthStr), &registriesAuth)
 	if err != nil {
-		return nil, fmt.Errorf("error parsing Secret: %s", err.Error())
-	}
-	//try to find an auth with the same registry name from the request
-	for _, auth := range registriesAuth {
-		if auth.Registry == registryName {
-			if err := auth.initDefaultValues(); err != nil {
-				return nil, err
-			}
-			return &auth, nil
-		}
+		sessionObj.Reporter.SendStatus(reporterlib.JobFailed, true, sessionObj.ErrChan)
+		glog.Errorf("in parseRegistryCommand: error: ", err.Error())
+		return fmt.Errorf("parseRegistryCommand failed with err %v", err)
 	}
-	//couldn't find auth with the full, check if there is an auth for the registry without the project name
-	regAndProject := strings.Split(registryName, "/")
-	if len(regAndProject) > 1 {
-		for _, auth := range registriesAuth {
-			if auth.Registry == regAndProject[0] {
-				if err := auth.initDefaultValues(); err != nil {
-					return nil, err
-				}
-				return &auth, nil
-			}
-		}
 
-	}
-	//no auth found for registry return a default one
-	auth := makeRegistryAuth(registryName)
-	return &auth, nil
+	return actionHandler.scanRegistry(&registryScan, sessionObj)
 }
 
-func (actionHandler *ActionHandler) getRegistryConfig(registryName string) (*registryScanConfig, string, error) {
-	configMap, err := actionHandler.k8sAPI.GetWorkload(armotypes.KubescapeNamespace, "ConfigMap", registryScanConfigmap)
-	// in case of an error or missing configmap, fallback to the deprecated namespace
-	if err != nil || configMap == nil {
-		configMap, err = actionHandler.k8sAPI.GetWorkload(armotypes.ArmoSystemNamespace, "ConfigMap", registryScanConfigmap)
+func (actionHandler *ActionHandler) testRegistryConnectivity(sessionObj *utils.SessionObj) error {
+	registryScan := NewRegistryScan(actionHandler.k8sAPI)
+
+	err := registryScan.parseRegistryFromCommand(sessionObj)
```

Comment:
This should be a generic function which loads the necessray fields from CM/secret/request body and return unified object